src/accordion-item/AccordionItem.js

```diff
@@ -0,0 +1,104 @@
+import Content from './components/Content';
+
+const AccordionItem = ({
+    animationDuration,
+    animationEasing,
+    children,
+    itemContainerClassName,
+    headingContainerClassName,
+    titleContainerClassName,
+    contentContainerClassName,
+    contentClassName,
+    iconClassName,
+    icon,
+    index,
+    isActive,
+    onItemToggle,
+    scrollIntoView,
+    toggleItemActive,
+    title,
+}) => {
+    const containerRef = useRef();
+    const wasActive = usePrevious(isActive);
+
+    useEffect(() => {
+        let scrollTimeout;
+
+        if (wasActive !== isActive && typeof wasActive !== 'undefined') {
+            onItemToggle(index, { active: isActive });
+        }
+
+        if (!wasActive && isActive && scrollIntoView) {
+            scrollTimeout = setTimeout(() => {
+                scrollIntoViewIfNeeded(containerRef.current, {
+                    block: 'nearest',
+                    behavior: 'smooth',
+                    scrollMode: 'if-needed',
+                });
+            }, animationDuration);
+        }
+
+        return () => {
+            clearTimeout(scrollTimeout);
+        };
+    }, [animationDuration, containerRef, index, isActive, onItemToggle, scrollIntoView, wasActive]);
+
+    return (
+        <div
+            ref={ containerRef }
+            data-testid={ `accordion-item-${index}` }
+            className={ classNames('react-accordion-item', { active: isActive }, itemContainerClassName) }>
+            <Heading
+                index={ index }
+                isActive={ isActive }
+                icon={ icon }
+                iconClassName={ iconClassName }
+                toggleItemActive={ toggleItemActive }
+                containerClassName={ headingContainerClassName }
+                titleContainerClassName={ titleContainerClassName }
+                title={ title } />
+            <Content
+                isActive={ isActive }
+                animationDuration={ animationDuration }
+                animationEasing={ animationEasing }
+                contentClassName={ contentClassName }
+                containerClassName={ contentContainerClassName }>
+                { children }
+            </Content>
+        </div>
+    );
+};
+
+AccordionItem.propTypes = {
+    animationDuration: PropTypes.number,
+    animationEasing: PropTypes.string,
+    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
+    itemContainerClassName: PropTypes.string,
+    contentContainerClassName: PropTypes.string,
+    contentClassName: PropTypes.string,
+    headingContainerClassName: PropTypes.string,
+    titleContainerClassName: PropTypes.string,
+    icon: PropTypes.any,
+    iconClassName: PropTypes.string,
+    index: PropTypes.number,
+    isActive: PropTypes.bool,
+    onItemToggle: PropTypes.func,
+    scrollIntoView: PropTypes.bool,
+    toggleItemActive: PropTypes.func,
+    title: PropTypes.any.isRequired,
+};
```

Comment:
Something that might help you out, is to comment what props are passed by the Accordion parent. For example:
```js
AccordionItem.propTypes = {
    /* Accordion */
    index: PropTypes.number,
    isActive: PropTypes.bool,
    ...
    /* External */
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.func]).isRequired,
    title: PropTypes.any.isRequired,
    ...
};
```

Reply:
This makes sense, I will do that 👍